lumicks/pylake/kymotracker/kymoline.py

```diff
@@ -338,32 +338,50 @@ def estimate_diffusion(self, method, max_lag=None):
-        estimates to use. See [1] for more information on this.
+        the estimate worse.
+
+        There are two ways around this. Either you determine an optimal number of points to use
+        in the estimation procedure (ols) [1] or you take into account the covariances present in
+        the mean squared difference estimates (gls) [2].
+
+        The standard error returned by this method is based on [2].
 
         Note that this estimation procedure should only be used for pure diffusion in the absence
         of drift.
 
-        1) Michalet, X., & Berglund, A. J. (2012). Optimal diffusion coefficient estimation in
-        single-particle tracking. Physical Review E, 85(6), 061916.
-        2) Bullerjahn, J. T., von Bülow, S., & Hummer, G. (2020). Optimal estimates of
-        self-diffusion coefficients from molecular dynamics simulations. The Journal of Chemical
-        Physics, 153(2), 024116.
-
         Parameters
         ----------
         method : str
-            Must be "ols".
-            - "ols" : Ordinary least squares [1]. Includes standard error estimate based on [2].
+            Valid options are "ols" and "gls".
+
+            - "ols" : Ordinary least squares [1]. Determines optimal number of lags.
+            - "gls" : Generalized least squares [2]. Takes into account covariance matrix (slower).
         max_lag : int (optional)
-            Number of lags to include. When omitted, the method calculates an appropriate number of
-            lags.
+            Number of lags to include. When omitted, the method will choose an appropriate number
+            of lags to use.
+            When the method chosen is "ols" an optimal number of lags is estimated as determined by
+            [1]. When the method is set to "gls" all lags are included.
+
+        References
+        ----------
+        .. [1] Michalet, X., & Berglund, A. J. (2012). Optimal diffusion coefficient estimation in
+               single-particle tracking. Physical Review E, 85(6), 061916.
+        .. [2] Bullerjahn, J. T., von Bülow, S., & Hummer, G. (2020). Optimal estimates of
+               self-diffusion coefficients from molecular dynamics simulations. The Journal of Chemical
+               Physics, 153(2), 024116.
         """
-        if method != "ols":
-            raise ValueError('Invalid method selected. Method must be "ols"')
+        if method not in ("gls", "ols"):
+            raise ValueError('Invalid method selected. Method must be "gls" or "ols"')
 
         frame_idx, positions = np.array(self.time_idx, dtype=int), np.array(self.position)
-        max_lag = max_lag if max_lag else determine_optimal_points(frame_idx, positions)[0]
+        max_lag = (
+            max_lag
+            if max_lag
+            else determine_optimal_points(frame_idx, positions)[0]
+            if method == "ols"
+            else len(frame_idx)
+        )
+
```

Comment:
oof, I love a good ternary but this may be too much. consider seeing if the usual `if-else` block would be more readable. or at the very least maybe an extra set of `()` for the first `else`